### ISSUE TITLE
Update set up configuration page to v2 Standards

### ISSUE THREAD

**Issue by Nopzen** [redux-json-api/redux-json-api#164]
This page is still running the V1 setup, it is referencing the old setup api's:

`setEndpointHost`, `setEndpointPath`, `setAccessToken`, `setHeader` and `setHeader`

These are all been deprecated in favour of the `setAxiosConfig` api.

https://github.com/redux-json-api/redux-json-api/blob/master/docs/set-up-configure.md

**Comment by egeriis** [comment]
Let's get a PR :tada:

**Comment by Nopzen** [comment]
Ill try to get to it tonight, some RL stuff came in the way.

**Comment by egeriis** [comment]
@Nopzen Had a chance to look at this?

**Comment by Nopzen** [comment]
@egeriis Sorry no. I'll try to take a look asap, i got to open some codebase today where i use this.

**Comment by Nopzen** [comment]
@egeriis so i can see i just use a very basic example:

```
store.dispatch(setAxiosConfig({
    baseURL: 'http://localhost:3000/'
}));
```

And as i can see out of the code here:
https://github.com/redux-json-api/redux-json-api/blob/master/src/jsonapi.js#L194
```
[API_SET_AXIOS_CONFIG]: (state, { payload: axiosConfig }) => {
    return imm(state).set(['endpoint', 'axiosConfig'], axiosConfig).value();
  },
```

We are basically on the key endpoint setting the given Axios config given as param to the action.

My assumtion here is we can pass a config as the given example of the Axios repo: https://github.com/axios/axios#axioscreateconfig? So supporting: `baseurl`, `timeout` & `headers`?

Also i wonder why is it stored on a key called `endpoint` should it not just be `axiosConfig` ??

**Comment by egeriis** [comment]
Yeah, the configuration you provide are directly proxied to axios. 

> Also i wonder why is it stored on a key called `endpoint` should it not just be `axiosConfig` ??

Legacy. We used to have endpoint and base path specified here. Feel free to move it. Should not break anything.

**Comment by Nopzen** [comment]
So okay I'am back on track now.

So i, working on this PR again and for now i will update the documentation on how to get going, and setting up the axios config.

And on this comment form @egeriis 
> Legacy. We used to have endpoint and base path specified here. Feel free to move it. Should not break anything.

I investigated it broke a lot by removing the `endpoints` key, i expected this, but I've refactored it to a working state again, and are no building tests for this use case, meanwhile i explored this refactor i discovered we didn't have test cases for the redux-thunk actions, these are the one i  will create a bare minimum test cases that covers my refactor.

The test setup will include as of now instalation of package Chai, and moxios (mocking axios), i hope to have a pr soon.

Sorry for the delay of things.
